cpp/src/svm/svc_impl.h

```diff
@@ -119,42 +123,75 @@ void svcFit(const cumlHandle &handle, math_t *input, int n_rows, int n_cols,
  * @param [in] model SVM model parameters
  * @param [out] preds device pointer to store the predicted class labels.
  *    Size [n_rows]. Should be allocated on entry.
+ * @param [in] buffer_size size of temporary buffer in MiB
  */
 template <typename math_t>
 void svcPredict(const cumlHandle &handle, math_t *input, int n_rows, int n_cols,
                 MLCommon::Matrix::KernelParams &kernel_params,
-                const svmModel<math_t> &model, math_t *preds) {
+                const svmModel<math_t> &model, math_t *preds,
+                math_t buffer_size) {
   ASSERT(n_cols == model.n_cols,
          "Parameter n_cols: shall be the same that was used for fitting");
   // We might want to query the available memory before selecting the batch size.
   // We will need n_batch * n_support floats for the kernel matrix K.
-#define N_PRED_BATCH 4096
+  const int N_PRED_BATCH = 4096;
```

Comment:
Would it make sense for this to be a function argument w/ a default value? If I'm not mistaken, I'm taking this to be the max possible batch size when `n_rows < N_PRED_BATCH`

Reply:
> Ideally, we should have a deterministic rule to set N_PRED_BATCH to a value which gives good perf. 

Sounds good.